pkg/controllers/monitoring/monitoring-stack/conditions.go

```diff
@@ -0,0 +1,121 @@
+	cm := make(map[stack.ConditionType]*stack.Condition)
+	cm[stack.Available] = &stack.Condition{
+		Type:               stack.Available,
+		Status:             stack.ConditionUnknown,
+		LastTransitionTime: metav1.Now(),
+	}
+
+	cm[stack.Reconciled] = &stack.Condition{
+		Type:               stack.Reconciled,
+		Status:             stack.ConditionUnknown,
+		LastTransitionTime: metav1.Now(),
+	}
+	return conditionsMap{conditions: cm}
+}
+
+// UpdateAvailable gets existing "Available" condition and updates its parameters
+// based on the Prometheus "Available" condition
+func (cm *conditionsMap) UpdateAvailable(prometheusConditions []monv1.PrometheusCondition, generation int64) {
+	ac := cm.conditions[stack.Available]
+	ac.ObservedGeneration = generation
+
+	prometheusAvailable := getPrometheusCondition(prometheusConditions, monv1.PrometheusAvailable)
+
+	if prometheusAvailable == nil {
+		ac.Status = stack.ConditionFalse
+		ac.Reason = prometheusNotAvailable
+		ac.Message = cannotReadPrometheusConditions
+		ac.LastTransitionTime = metav1.Now()
+		return
+	}
+
+	if prometheusAvailable.Status != monv1.PrometheusConditionTrue {
+		ac.Status = stack.ConditionFalse
+		ac.Reason = prometheusNotAvailable
+		ac.Message = prometheusAvailable.Message
+		ac.LastTransitionTime = metav1.Now()
+		return
+	}
+
+	ac.Status = stack.ConditionTrue
+	ac.Reason = monitoringStackAvailable
+	ac.Message = "Monitoring Stack is available"
+}
+
+// UpdateReconciled updates "Reconciled" conditions based on the provided error value and
+// Prometheus "Reconciled" condition
+func (cm *conditionsMap) UpdateReconciled(prometheusConditions []monv1.PrometheusCondition, generation int64, err error) {
+	rc := cm.conditions[stack.Reconciled]
+	rc.ObservedGeneration = generation
+
+	if err != nil {
+		rc.Status = stack.ConditionFalse
+		rc.Message = err.Error()
+		rc.Reason = "FailedToReconcile"
+		rc.LastTransitionTime = metav1.Now()
+		return
+	}
+	prometheusReconciled := getPrometheusCondition(prometheusConditions, monv1.PrometheusReconciled)
+
+	if prometheusReconciled == nil {
+		rc.Status = stack.ConditionFalse
+		rc.Reason = prometheusNotReconciled
+		rc.Message = cannotReadPrometheusConditions
+		rc.LastTransitionTime = metav1.Now()
+		return
+	}
+
+	if prometheusReconciled.Status != monv1.PrometheusConditionTrue {
+		rc.Status = stack.ConditionFalse
+		rc.Reason = prometheusNotReconciled
+		rc.LastTransitionTime = metav1.Now()
+		rc.Message = prometheusReconciled.Message
+		return
+	}
+
+	rc.Status = stack.ConditionTrue
+	rc.Reason = monitoringStackReconciled
+	rc.Message = "Monitoring Stack successfully reconciled"
+}
+
+func (cm *conditionsMap) ToSlice() []stack.Condition {
+	s := make([]stack.Condition, 0, len(cm.conditions))
+	for _, v := range cm.conditions {
+		s = append(s, *v)
+	}
+	return s
+}
+
+func getPrometheusCondition(prometheusConditions []monv1.PrometheusCondition, t monv1.PrometheusConditionType) *monv1.PrometheusCondition {
```

Comment:
The function could return a value + error instead of a pointer and a non-nil error would mean that the condition type couldn't be found. This would be more idiomatic :)

```suggestion
func getPrometheusCondition(prometheusConditions []monv1.PrometheusCondition, t monv1.PrometheusConditionType) (monv1.PrometheusCondition, error) {
```

Reply:
Updated. Thanks